### PR TITLE
New build process

### PR DESCRIPTION
**Pull request by vforge**
- Move all the source files to `source` directory.

- Update READMEs everywhere.

- For now (until we have private npm account or artifactory) we cannot publish the package, so the output files need to stay in this repo.

- I do not want the client to download X libraries that are not needed in order to use the RDS (like Ember is needed for the DS itself), so the process is compiling all the files into ES5 and you only need React, ReactDOM and PropTypes in order to use them. The client shouldn't care if an RDS component needs lodash or anything else.

- For the reason above I do not want to have a post-install process. In the future, the build should happen only before the package is published.

NOTE: This PR does not contain build files, because I wanted it to be easier to read. Once it's clear, I'll add the result of `yarn build`.

### CODE REVIEW

**Comment by JoshuaRogan** [comment]
I think nearly all of these files should be generated on `prepublishOnly` or something similar. It makes it very hard to write code with all of this extra code.

**Comment by mklucsarits** [comment]
@v Can you explain a bit more what's going on here? Like what the extra code is doing and how it's generated?

File: components/BannerNotifications.js [hunk 1]
```diff
@@ -0,0 +1,366 @@
+'use strict';
+
+function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
```
[the remark this answers is not in the document]
Comment (vforge) [reply components/BannerNotifications.js]: yes!

**Comment by vforge** [comment]
@JoshuaRogan Unfortunately because of our licensing we won't publish this package to npm.js. There are other ideas floating around, but for now everything stays on GH.

@mklucsarits Sure, let me add READMEs and some description during the weekend and we'll get back to this PR next week.

**Comment by JoshuaRogan** [comment]
I believe you can run it during npm `install` without it being on npm. https://docs.npmjs.com/misc/scripts

**Comment by vforge** [comment]
> Don't use install. Use a .gyp file for compilation, and prepublish for anything else. You should almost never have to explicitly set a preinstall or install script. If you are doing this, please consider if there is another option. The only valid use of install or preinstall scripts is for compilation which must be done on the target architecture.

https://docs.npmjs.com/misc/scripts#best-practices

**Comment by vforge** [comment]
I'll in the process of updating docs/readmes so it's clear why it's like that right now.

**Comment by vforge** [comment]
@mklucsarits and @JoshuaRogan I've updated PR and added a description. Hopefully, it's clearer now.

**Comment by vforge** [comment]
@mklucsarits automated dismiss, sorry about that! 
screenshot 2018-10-02 15 38 56

I turned it off

**Comment by JoshuaRogan** [comment]
@vforge Can we use artifactory to host the content? It really sucks having the build files in the repo.

**Comment by vforge** [comment]
@JoshuaRogan I'll work on that.

**Pull request merged**